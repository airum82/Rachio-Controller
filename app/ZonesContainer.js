import React from 'react';
import Zone from './Zone';

const ZonesContainer = ({ zones, history }) => {
  return (
    <div className="zones-container">
      <button onClick={history.goBack}>back</button>
      { zones.map((zone, index) => {
        return (
          <Zone 
            name={zone.name}
            id={zone.id}
            key={index}
            image={zone.imageUrl}
          />
        )
      })}
    </div>
  )
}

export default ZonesContainer;