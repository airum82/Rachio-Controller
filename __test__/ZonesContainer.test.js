import React from 'react';
import ZonesContainer from '../app/ZonesContainer';
import { shallow } from 'enzyme';

describe('ZonesContainer', () => {
  it('should match snapshot upon render', () => {
    const mockZones = [{}, {}];
    const mockSelectZone = jest.fn();
    const wrapper = shallow(<ZonesContainer 
      zones={mockZones}
      selectZone={mockSelectZone}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})