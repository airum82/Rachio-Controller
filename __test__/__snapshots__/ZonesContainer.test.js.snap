// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZonesContainer should match snapshot upon render 1`] = `
<div
  className="zones-container"
>
  <Zone
    key="0"
    selectZone={[MockFunction]}
  />
  <Zone
    key="1"
    selectZone={[MockFunction]}
  />
</div>
`;
